src/component/structure-component.ts

```diff
@@ -354,8 +362,16 @@ class StructureComponent extends Component {
     return bb
   }
 
-  getCenterUntransformed (sele: string): Vector3 {
-    if (sele && typeof sele === 'string') {
+  setCenterMode(mode: CenterMode) {
+    this.centerMode = mode
+    this.updateMatrix()
+  }
+
+  // Structure-component center can be based on the selection or the whole structure.
+  // Caller can override use of selection-center mode by passing useCenterMode = false.
+  getCenterUntransformed (useCenterMode: boolean = true): Vector3 {
```

Comment:
This removes the option to pass an explicit selection through to this method. `getCenterUntransformed(sele: string)`, replacing it with a boolean arg. 

I think this will break `Component.getCenter` which passes its own args through directly to this method, and is used in a number of examples.

I couldn't see a call to `getCenterUntransformed(true|false)` in this PR, but if you're using it in your code could you replace it with explicit calls to the selection string you want?

Reply:
Good point. I'll fix this so it accepts an optional selection as the first arg (I think it should be optional -- not all callers pass it today), and uses that instead of the current selection when passed in.